protected/models/forms/project/Create.php

```diff
@@ -0,0 +1,61 @@
+<?php
+declare(strict_types=1);
+
+namespace prime\models\forms\project;
+
+use prime\models\ar\Project;
+use prime\objects\enums\ProjectVisibility;
+use Yii;
+use yii\base\Model;
+use yii\helpers\ArrayHelper;
+use yii\validators\RangeValidator;
+use yii\validators\RequiredValidator;
+use yii\validators\SafeValidator;
+use yii\validators\StringValidator;
+use yii\validators\UniqueValidator;
+use function iter\filter;
+
+class Create extends Model
+{
+    public string $title = '';
+    public ProjectVisibility $visibility;
+    public null|int $base_survey_eid = null;
+
+    public function __construct()
+    {
+        parent::__construct([]);
+        $this->visibility = ProjectVisibility::public();
+    }
+
+
+    public function dataSurveyOptions(): array
+    {
+        $existing = Project::find()->select('base_survey_eid')->indexBy('base_survey_eid')->column();
+
+        $surveys = filter(function ($details) use ($existing) {
+            return (isset($this->base_survey_eid) && $this->base_survey_eid == $details['sid']) || !isset($existing[$details['sid']]);
+        }, app()->limesurveyDataProvider->listSurveys());
+
+        $result = ArrayHelper::map($surveys, 'sid', function ($details) {
+            return $details['surveyls_title'] . (($details['active'] == 'N') ? " (INACTIVE)" : "");
+        });
+
+        return $result;
+    }
+
+    public function formName(): string
+    {
+        return 'Project';
```

Comment:
Isn't it an idea to pass a `new Project` in the constructor so we can extract the formName there? Or some other solution so we don't need to hardode the `formName`